<launch>
  <arg name="data_path" default="/root/rosbags/LEMONMapping_data/EastTeachingBuilding/drone"/>
  <arg name="loop_path" default="/root/rosbags/LEMONMapping_data/EastTeachingBuilding/loops.txt"/>
  <arg name="origin_pose_name" default="pose"/>
  <arg name="pose_opt_path" default="/root/rosbags/LEMONMapping_data/EastTeachingBuilding/drone"/>
  <arg name="robot_num" default="3"/>
  <arg name="rviz" default="false"/>

  <node pkg="voxel_slam" type="voxel_offline_pgo" name="voxel_offline_pgo" output="screen">
    <param name="data_path" type="string" value="$(arg data_path)"/>
    <param name="loop_path" type="string" value="$(arg loop_path)"/>
    <param name="origin_pose_name" type="string" value="$(arg origin_pose_name)"/>
    <param name="pose_opt_path" type="string" value="$(arg pose_opt_path)"/>
    <param name="robot_num" type="int" value="$(arg robot_num)"/>
  </node>

  <group if="$(arg rviz)">
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find voxel_slam)/rviz_cfg/back.rviz" output="screen"/>
  </group>
</launch>
